Source/ConstraintConstantTarget.swift

```diff
@@ -91,53 +95,73 @@ extension ConstraintConstantTarget {
                     return value.y
                 case .width, .height, .notAnAttribute:
                     return 0.0
-                }
+                #if swift(>=5.0)
+                @unknown default:
+                    return 0.0
+                #endif
+            }
             #endif
         }
         
         if let value = self as? ConstraintInsets {
             #if os(iOS) || os(tvOS)
                 switch layoutAttribute {
-                case .left, .leftMargin, .centerX, .centerXWithinMargins:
+                case .left, .leftMargin:
                     return value.left
-                case .top, .topMargin, .centerY, .centerYWithinMargins, .lastBaseline, .firstBaseline:
+                case .top, .topMargin, .firstBaseline:
                     return value.top
                 case .right, .rightMargin:
                     return -value.right
-                case .bottom, .bottomMargin:
+                case .bottom, .bottomMargin, .lastBaseline:
                     return -value.bottom
                 case .leading, .leadingMargin:
                     return (ConstraintConfig.interfaceLayoutDirection == .leftToRight) ? value.left : value.right
                 case .trailing, .trailingMargin:
                     return (ConstraintConfig.interfaceLayoutDirection == .leftToRight) ? -value.right : -value.left
+                case .centerX, .centerXWithinMargins:
+                    return (value.left - value.right) / 2
```

Comment:
Seems like this should be `(value.left + value.right) / 2` – the average, not the difference...?

Reply:
Ohh no I see, the bottom and right values are inverted sign, so this *is* doing addition.  Got it.